scripts/Muon/GUI/ElementalAnalysis2/context/ea_group_context.py

```diff
@@ -0,0 +1,120 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from Muon.GUI.ElementalAnalysis2.ea_group import EAGroup
+from mantidqt.utils.observer_pattern import GenericObservable
+
+INVALID_STRINGS_FOR_GROUP_NAMES = ["EA_Rebinned_Fixed", "EA_Rebinned_Variable"]
+
+
+def get_default_grouping(loadedData):
+    """this creates the first set of groups listed in the grouping table of the Elemental Analysis GUI
+        For single workspace names the detector is found by taking everything after ; in the name
+        For example : 2695; Detector 1 --> Detector 1
+        For co-added workspaces the detector is found by taking everything before
+        For example : Detector 1_2695-2686 --> Detector 1
+    """
+    groups = []
+    run_list = loadedData.get_parameter("run")
+    for run_item in run_list:
+        for workspace in loadedData.get_data(run=run_item)["workspace"]:
+            group_name = str(workspace)
+            print(group_name)
```

Comment:
Do we still want this print here?

Reply:
No it should not be there thanks for spotting that